content/_code-samples/airgapped-wallet/py/airgapped-wallet.py

```diff
@@ -0,0 +1,218 @@
+                                            "\n    Enter Ledger Sequence: "))
+
+                sign_transaction(_xrp_amount=amount,
+                                 _destination=destination,
+                                 _ledger_seq=ledger_sequence,
+                                 _wallet_seq=wallet_sequence,
+                                 password=password
+                                 )
+
+                del destination, amount, wallet_sequence, ledger_sequence
+
+            if ask == 2:
+                _pub, _seed = create_wallet()
+
+            if ask == 3:
+                with open(get_path("/Wallet/public.txt"), "r") as f:
+                    print(f"\n  Wallet Address: {f.read()}")
+
+                image = Image.open(get_path("/Wallet/public.png"))
+                image.show()
+
+    else:
+        # If the Wallet's folder does not exist, create one and store wallet data (encrypted private key, encrypted seed, account address)
+
+        # If the Wallet's directory exists but files are missing, delete it and generate a new wallet
+        if os.path.exists(File):
+            confirmation = input(f"We've detected missing files on {File}, would you like to delete your wallet's credentials & generate new wallet credentials? (YES/NO):")
+            if confirmation == "YES":
+                confirmation_1 = input(f"All wallet credentials will be lost if you continue, are you sure? (YES/NO): ")
+                if confirmation_1 == "YES":
+                    shutil.rmtree(File)
+                else:
+                    print("Aborted: Wallet credentials are still intact")
+                    return 0
+            else:
+                print("- Wallet credentials are still intact")
+                return 0
+
+        os.makedirs(File)
+
+        pub, seed = create_wallet()
+
+        img = qrcode.make(pub)
+        img.save(get_path("/Wallet/public.png"))
+
+        print("\nCreating a brand new Wallet, please enter a new password")
+        password = str(input("\n        Enter Password: "))
+        salt = os.urandom(16)
+
+        with open(get_path("/Wallet/salt.txt"), "wb") as f:
+            f.write(salt)
+
+        kdf = PBKDF2HMAC(
+            algorithm=hashes.SHA256(),
+            length=32,
+            iterations=100000,
+            salt=salt
+        )
+
+        key = base64.urlsafe_b64encode(kdf.derive(bytes(password.encode())))
+
+        crypt = Fernet(key)
+
+        priv = crypt.encrypt(bytes(seed, encoding='utf-8'))
+        seed = crypt.encrypt(bytes(seed, encoding='utf-8'))
+
+        with open(get_path("/Wallet/seed.txt"), "w") as f:
+            f.write(seed.hex())
+
+        with open(get_path("/Wallet/private.txt"), "w") as f:
+            f.write(priv.hex())
+
+        with open(get_path("/Wallet/public.txt"), "w") as f:
+            f.write(pub)
+
+        openimg = Image.open(get_path("/Wallet/public.png"))
+        openimg.show()
+
+        print("Re-run this script!")
```

Comment:
```suggestion
        print("Finished generating an account.")
        print("Re-run this script to use this account and credentials!")
```

Reply:
Adding a status message to let the user know that running it for the first time did accomplish something which will make the 2nd run do something different.

Reply:
(If we want them to re-run it though, should the code just jump into the main loop logic directly after completing the first step?)

Reply:
Also adding in instructions to fund the account using the BitHomp faucet site since that's an important step we need the user to do.